app/harvesters/hathi_harvester.rb

```diff
@@ -0,0 +1,72 @@
+require 'net/http'
+require 'rubygems/package'
+require 'zlib'
+
+
+class HathiHarvester < Krikri::Harvesters::MarcXMLHarvester
+
+  private
+
+  def tarball_source(&block)
+    if File.exists?(uri)
+      File.open(uri, &block)
+    else
+      download_url(get_latest_tarball_url, &block)
+    end
+  end
+
+  def each_collection
+    tarball_source do |tar_file|
+      each_tar_entry(tar_file) do |entry|
+        yield(entry)
+      end
+    end
+  end
+
+  def get_latest_tarball_url
+    html = Net::HTTP.get(URI(uri))
+    parsed_html = Nokogiri::HTML(html)
+
+    file_names = parsed_html.xpath('//a[starts-with(@href, "dpla")]').map { |a| a.get_attribute("href") }
+    latest_file = file_names.sort.reverse.first
+
+    URI.join(uri, latest_file)
+  end
+
+  def download_url(url, &block)
+    Net::HTTP.start(url.host, url.port) do |http|
+      request = Net::HTTP::Get.new(url.request_uri)
+
+      file = Tempfile.new('hathi', Dir.tmpdir, :encoding => 'ASCII-8BIT')
+
+      http.request(request) do |resp|
+        resp.read_body do |seg|
+          file.write(seg)
+        end
+      end
+
+      file.close
+
+      if block
```

Comment:
@marktriggs is there a reason we went with `if block` and `block.call(file)` rather than `if block_given?` and `yield file` here?

I think these behave differently under various circumstances (definitely `if block` executes when `block` is some non-closure object), and I'm wondering if that's intentional.